src/main/java/ch/njol/skript/expressions/ExprYawPitch.java

```diff
@@ -35,118 +39,182 @@
-	"set {_yaw} to yaw of player",
-	"set {_p} to pitch of target entity"})
-@Since("2.0, 2.2-dev28 (vector yaw/pitch)")
+		"set {_yaw} to yaw of player",
+		"set {_p} to pitch of target entity",
+		"add 90 to yaw of player # Rotates the player counter-clockwise"})
+@Since("2.0, 2.2-dev28 (vector yaw/pitch), INSERT VERSION (changers)")
 public class ExprYawPitch extends SimplePropertyExpression<Object, Number> {
 
 	static {
-		register(ExprYawPitch.class, Number.class, "(0¦yaw|1¦pitch)", "locations/vectors");
+		register(ExprYawPitch.class, Number.class, "(:yaw|pitch)", "entities/locations/vectors");
 	}
 
+	// For non-Paper versions lower than 1.19, changing the rotation of an entity is not supported for players.
+	private static final boolean SUPPORTS_PLAYERS = Skript.isRunningMinecraft(1, 19) && Skript.getServerPlatform() == ServerPlatform.BUKKIT_PAPER;
+
 	private boolean usesYaw;
 
 	@Override
-	public boolean init(final Expression<?>[] exprs, final int matchedPattern, final Kleenean isDelayed, final ParseResult parseResult) {
-		usesYaw = parseResult.mark == 0;
-		return super.init(exprs, matchedPattern, isDelayed, parseResult);
+	public boolean init(Expression<?>[] expressions, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
+		usesYaw = parseResult.hasTag("yaw");
+		return super.init(expressions, matchedPattern, isDelayed, parseResult);
 	}
 
 	@Override
-	public Number convert(final Object object) {
-		if (object instanceof Location) {
-			Location l = ((Location) object);
-			return usesYaw ? convertToPositive(l.getYaw()) : l.getPitch();
+	public Number convert(Object object) {
+		if (object instanceof Entity) {
+			Location location = ((Entity) object).getLocation();
+			return usesYaw
+					? toSkriptYaw(location.getYaw())
+					: toSkriptPitch(location.getPitch());
+		} else if (object instanceof Location) {
+			Location location = (Location) object;
+			return usesYaw
+					? toSkriptYaw(location.getYaw())
+					: toSkriptPitch(location.getPitch());
 		} else if (object instanceof Vector) {
-			Vector vector = ((Vector) object);
-			if (usesYaw)
-				return VectorMath.skriptYaw(VectorMath.getYaw(vector));
-			return VectorMath.skriptPitch(VectorMath.getPitch(vector));
+			Vector vector = (Vector) object;
+			return usesYaw
+					? toSkriptYaw(VectorMath.getYaw(vector))
+					: toSkriptPitch(VectorMath.getPitch(vector));
 		}
 		return null;
 	}
 
-	@SuppressWarnings({"null"})
 	@Override
-	public Class<?>[] acceptChange(final ChangeMode mode) {
-		if (mode == ChangeMode.SET || mode == ChangeMode.ADD || mode == ChangeMode.REMOVE)
+	public Class<?>[] acceptChange(ChangeMode mode) {
+		if (getExpr().getReturnType().isAssignableFrom(Player.class) && !SUPPORTS_PLAYERS)
+			return null;
+		if (mode == ChangeMode.SET || mode == ChangeMode.ADD || mode == ChangeMode.REMOVE) {
 			return CollectionUtils.array(Number.class);
+		} else if (mode == ChangeMode.RESET) {
+			return new Class[0];
+		}
 		return null;
```

Comment:
This can be
```java
switch (mode) {
		case SET:
		case ADD:
		case REMOVE:
		case RESET:
				return CollectionUtils.array(Number.class);
		default:
				return null;
}
```

Reply:
I personally find that a little abusive of the switch-case system. I also separated the RESET mode to return an empty array to follow the guidelines present in the Javadocs of that method, which state something like "DELETE and RESET may return an empty array to accept the change".

Reply:
That's for Expression get methods, this must return null if it doesn't support that changer. I'll update that comment to address it's specifics.

Reply:
switches are better for code readability and returning null allows Skript to provide a suggestion error message, optionally.